### ISSUE TITLE
BUG: Elbow Arrow ALT+drag duplication results in fuzzy/confusing UX

### ISSUE THREAD

**Issue by zsviczian** [excalidraw/excalidraw#8435]
Alt+dragging a single elbow connector that is bound to at least one element results in a duplicated element right on top of the previous element. Dragging the arrow - to move it - has no effect.

When duplicating the new arrow should not carry its bindings. 

https://github.com/user-attachments/assets/cda2761a-b97a-4977-b013-d7e7f5cd098c

**Comment by zsviczian** [comment]
On a related issue, duplicating a non-elbow arrow loses its binding.

https://github.com/user-attachments/assets/5e14e589-f7d2-466e-ad4c-2c9e498c53a4

**Comment by dwelle** [comment]
@mtolmacs I believe this function is planned to be completely rewritten after you finish up with the math rewrites?

**Comment by dwelle** [comment]
> Alt+dragging a single elbow connector that is bound to at least one element results in a duplicated element right on top of the previous element. Dragging the arrow - to move it - has no effect.
> 
> When duplicating the new arrow should not carry its bindings.

Not sure what the behavior should be. FigJam does something similar to what we do. Miro doesn't even duplication at all.

**Comment by zsviczian** [comment]
If I only drag the arrow and not the bound elements, the arrow should be duplicated and unbound. I came across this today as I was drawing a flowchart. I had arrows with longer labels that I wanted to duplicate instead of draw and re-label. In the end CTRL+C CTRL+V did the trick...

**Comment by mtolmacs** [comment]
Indeed this is on my refactor list, but I haven't got around doing it yet. I'd be happy to see a reimplementation for this, but first we should describe what the intended behavior is for each duplication action.

@dwelle correct me if I'm wrong, but it should work like this:
- For each potential binding (start and end separately):
  - Is the bindable and the linear element are both duplicated
    - If ALT+drag duplication happened: CREATE BINDING TO THE DUPLICATION
    - If duplicate action happened: NO BINDING
  - Is the bindable or the linear element are not part of the duplication
    - If ALT+drag duplication happened: NO BINDING
    - If duplicate action happened: NO BINDING

Am I missing any case?

**Comment by zsviczian** [comment]
Currently, CTRL+D works as ALT+DRAG should.

I believe these are all the cases:
https://excalidraw.com/#room=d00aeaf60051c327ac88,WdnbhxpP7pxoMPOJ6Fqnzg

![image](https://github.com/user-attachments/assets/b50bee42-097f-4abf-8ff7-e395784bdd3a)

**Comment by mtolmacs** [comment]
I don't think we should concern ourselves with copy & paste, that is a different code path and copy&paste will not need any "fixing" after paste, it's either bringing over the binding in the element definition or not.

Also Alt+DRAG should not bind I think currently when only a linear element is duplicated but I defer to @dwelle.

**Comment by zsviczian** [comment]
I only mention copy paste, because it is mentioned at the head of the function:

```ts
export const fixBindingsAfterDuplication = (
  // There are three copying mechanisms: Copy-paste, duplication and alt-drag.
  // Only when alt-dragging the new "duplicates" act as the "old", while
  // the "old" elements act as the "new copy" - essentially working reverse
  // to the other two.
```

**Comment by zsviczian** [comment]
Currently, when you alt+drag an arrow the original arrow unbinds. That isn't desired behavior.

**Comment by dwelle** [comment]
> Indeed this is on my refactor list, but I haven't got around doing it yet. I'd be happy to see a reimplementation for this, but first we should describe what the intended behavior is for each duplication action.
> 
> @dwelle correct me if I'm wrong, but it should work like this:
> 
> * For each potential binding (start and end separately):
>   
>   * Is the bindable and the linear element are both duplicated
>     
>     * If ALT+drag duplication happened: CREATE BINDING TO THE DUPLICATION
>     * If duplicate action happened: NO BINDING
>   * Is the bindable or the linear element are not part of the duplication
>     
>     * If ALT+drag duplication happened: NO BINDING
>     * If duplicate action happened: NO BINDING
> 
> Am I missing any case?

yeah, makes sense

**Comment by imran1khan** [comment]
hi @zsviczian what's your discord username, can we talk please
